rust/wal3/src/backoff.rs

```diff
@@ -0,0 +1,55 @@
+use std::collections::hash_map::RandomState;
+use std::hash::BuildHasher;
+use std::time::{Duration, Instant};
+
+//////////////////////////////////////// ExponentialBackoff ////////////////////////////////////////
+
+pub struct ExponentialBackoff {
+    throughput_ops_sec: f64,
+    reserve_capacity: f64,
+    start: Instant,
+}
+
+impl ExponentialBackoff {
+    pub fn new(throughput_ops_sec: impl Into<f64>, reserve_capacity: impl Into<f64>) -> Self {
+        let throughput_ops_sec = throughput_ops_sec.into();
+        let reserve_capacity = reserve_capacity.into();
+        Self {
+            throughput_ops_sec,
+            reserve_capacity,
+            start: Instant::now(),
+        }
+    }
+
+    pub fn next(&self) -> Duration {
+        let elapsed = self.start.elapsed();
+        let recovery_window = Duration::from_micros(
+            (elapsed.as_micros() as f64 * self.throughput_ops_sec / self.reserve_capacity) as u64,
```

Comment:
what is the unit for `throughput_ops_sec` here? from the name it seems to be `ops/s` but from the calculation it looks like `ops/us`

Reply:
I've left a comment explaining.